src/SmartFormat/Core/Parsing/Parser.cs

```diff
@@ -280,7 +280,13 @@ public Format ParseFormat(string format, string[] formatterExtensionNames)
 
                             // Finish the last text item:
                             if (i != lastI) current.Items.Add(new LiteralText(Settings, current, lastI) {endIndex = i});
-                            lastI = i + 2;
+
+                            // Is this a unicode escape sequence?
+                            if (i + 1 < format.Length && format[i + 1] == 'u')
+                                lastI = i + 6; // \u0000 format
```

Comment:
Do you think it would be safe to do 
i += 6 and break here to save having to iterate over the numbers?

Reply:
Safe enough. It's the same as with other escaped characters. Illegal escape sequences throw when trying to unescape the literal by the formatter. I don't see an advantage already throwing in the parser.